Lisp/tests/syntax_test_lisp.lisp

```diff
@@ -0,0 +1,2865 @@
+;                                        ^^^^^^^^^^^^ meta.parens.lisp meta.parens.lisp meta.parens.lisp meta.parens.lisp
+;                                                    ^ meta.parens.lisp meta.parens.lisp meta.parens.lisp - meta.parens meta.parens meta.parens meta.parens
+;                                                     ^^ meta.parens.lisp meta.parens.lisp - meta.parens meta.parens meta.parens
+;                                                       ^ meta.parens.lisp - meta.parens meta.parens
+;^^^ keyword.declaration.variable.lisp
+;    ^ punctuation.section.parens.begin.lisp
+;      ^^^^^ meta.declaration.lisp variable.other.lisp
+;            ^ punctuation.section.parens.begin.lisp
+;             ^^^^^ variable.other.lisp
+;                   ^ punctuation.section.parens.begin.lisp
+;                    ^^^^ meta.function-call.lisp support.function.lisp
+;                         ^^^^^ variable.other.lisp
+;                              ^ punctuation.section.parens.end.lisp
+;                               ^ punctuation.section.parens.end.lisp
+;                                 ^ punctuation.section.parens.begin.lisp
+;                                  ^^^^^ variable.other.lisp
+;                                        ^ punctuation.section.parens.begin.lisp
+;                                         ^^^^ meta.function-call.lisp support.function.lisp
+;                                              ^^^^^ variable.other.lisp
+;                                                   ^^ punctuation.section.parens.end.lisp
+;                                                      ^^ punctuation.section.parens.end.lisp
+
+;########
+; LOOPS #
+;########
+
+(loop
+
+;;; name-clause
+
+  named name
+; ^^^^^ keyword.declaration.name.lisp
+;       ^^^^ entity.name.loop.lisp
+
+;;; initial-final-clause
+
+  initially ()
+; ^^^^^^^^^ keyword.control.flow.lisp
+;           ^^ meta.parens.lisp
+
+  finally (return (values evens odds))
+;^^^^^^^^^ meta.parens.lisp - meta.parens meta.parens
+;         ^^^^^^^^ meta.parens.lisp meta.parens.lisp - meta.parens meta.parens meta.parens
+;                 ^^^^^^^^^^^^^^^^^^^ meta.parens.lisp meta.parens.lisp meta.parens.lisp
+;                                    ^ meta.parens.lisp meta.parens.lisp - meta.parens meta.parens meta.parens
+;                                     ^ meta.parens.lisp - meta.parens meta.parens
+; ^^^^^^^ keyword.control.flow.lisp
+;         ^ punctuation.section.parens.begin.lisp
+;          ^^^^^^ keyword.control.flow.return.lisp
+;                 ^ punctuation.section.parens.begin.lisp
+;                  ^^^^^^ meta.function-call.lisp support.function.lisp
+;                         ^^^^^ variable.other.lisp
+;                               ^^^^ variable.other.lisp
+;                                   ^^ punctuation.section.parens.end.lisp
+
+;;; with-clause
+
+  with var1 :float = 10.5 and var2 = true
+; ^^^^ keyword.declaration.variable.lisp
```

Comment:
Someone better at Lisp than me should input whether this is right or if it's a context declaration. I would believe either.

Reply:
According to https://www.lispworks.com/documentation/lw70/CLHS/Body/m_loop.htm#loop, this is part of the `variable-clause` and looks rather like a variable declaration statement, followed by main-clauses, which describe looped statements.